### PR TITLE
Greps to check for commonly misplaced structures in closed turfs

### PR DESCRIPTION
**Pull request by Son-of-Space**
## About The Pull Request
Adds some checks for structures that are commonly misplaced inside walls. One of the more recent updatepaths also replaced a bunch of posters with windoors in walls so this will help find all those.

I also edited the one that checks for lattices to look for closed turfs, because we really don't want the structures we're looking for to be in any closed turfs anyways.

We don't want to look for all structures like this because things like transit tubes exist.

## Why It's Good For The Game

Even more quality control for our maps. We'll catch a lot of things before players do this way. : )

## Changelog

:cl:
code: Adds some greps to check for commonly misplaced structures in closed turfs
fix: Some objects stacked within closed turfs have been removed from those turfs.
/:cl:

### CODE REVIEW

**Comment by Son-of-Space** [comment]
Yosemama

File: tools/ci/check_grep.sh [hunk 1]
```diff
@@ -116,6 +116,21 @@ if grep -Pzo '"\w+" = \(\n[^)]*?/obj/structure/lattice[/\w]*?,\n[^)]*?/turf/clos
...
+fi;
+if grep -Pzo '"\w+" = \(\n[^)]*?/obj/structure/window[/\w]*?,\n[^)]*?/turf/closed[/\w]*?,\n[^)]*?/area/.+?\)' _maps/**/*.dmm;	then
+	echo
+    echo "ERROR: found a window stacked within a wall, please remove it."
+    st=1
+fi;
+if grep -Pzo '"\w+" = \(\n[^)]*?/obj/machinery/door/airlock[/\w]*?,\n[^)]*?/turf/closed[/\w]*?,\n[^)]*?/area/.+?\)' _maps/**/*.dmm;	then
+	echo
+    echo "ERROR: found an airlock stacked within a wall, please remove it."
+    st=1
```
Comment (Fikou) [inline tools/ci/check_grep.sh:132]: dude that shit is hilarious come on

**Comment by Maurukas** [comment]
Can't abandoned airlock helpers generate /turf/closed/wall underneath their airlock when they initialize?

**Comment by Son-of-Space** [comment]
> Can't abandoned airlock helpers generate /turf/closed/wall underneath their airlock when they initialize?

Yes, but those are intentional and shouldn't trigger these greps as the greps check before these spawners run through their code. These greps will check for unintentional ones or ones that deviate from our standards.

**Pull request merged**